### ISSUE TITLE
Recovery Partition breaks xVHDFile

### ISSUE THREAD

**Issue by Chirishman** [PowerShell/xHyper-V#134]
**Details of the scenario you tried and the problem that is occurring:**

When trying to inject files via DSC into a vhdx that contains a recovery partition as well as a primary partition the resource fails with an incredibly unhelpful message.

```
Cannot find drive. A drive with the name '
    + CategoryInfo          : ObjectNotFound: (:) [], CimException
    + FullyQualifiedErrorId : DriveNotFound,Microsoft.PowerShell.Commands.JoinPathCommand
    + PSComputerName        : FAKENAME
 
Cannot bind argument to parameter 'Path' because it is null.
    + CategoryInfo          : InvalidData: (:) [], CimException
    + FullyQualifiedErrorId : ParameterArgumentValidationErrorNullNotAllowed,Microsoft.PowerShell.Commands.TestPathCommand
    + PSComputerName        : FAKENAME
```

This is caused by the existing implementation's assumption that there will be only one partition. There appear to have been two attempts to fix this (by allowing specification of a partition number) which fizzled.

**The DSC configuration that is using the resource (as detailed as possible):**

```
xVhdFile BootstrapPayload {
    VHDPath = $DiskPath
    FileDirectory = @(
        MSFT_xFileDirectory {
            SourcePath = "C:\Fakepath\PretendMetaconfig.mof"
            DestinationPath = '\Windows\System32\Configuration\Metaconfig.mof'
            Type = 'File'
        }
}
```

**Version of the Operating System and PowerShell the DSC Target Node is running:**

ALL

**Version of the DSC module you're using, or 'dev' if you're using current dev branch:**

ALL

**Comment by Chirishman** [comment]
@joeyaiello Who  is maintaining this repo?

**Comment by johlju** [comment]
https://github.com/PowerShell/DscResources/blob/master/Maintainers.md

Look like it is @BrianFarnhill that are maintainer for this repository.

**Comment by Chirishman** [comment]
@johlju his last activity in this repo was June 27th and he was @ mentioned here almost a month ago and hasn't responded. I've poked him, @kwirkykat and @PowerShellTeam on Twitter and Gitter.im now with no response.

**Comment by Chirishman** [comment]
@PlagueHO I just saw your request to be tagged in the issue

**Comment by BrianFarnhill** [comment]
@Chirishman Dude, I'm sorry I missed this. Seems that my twitter notifications haven't been hitting m I y 
phone for some reason (I wondered why things got quiet!). I haven't been doing as much with GitHub over the last month, lots of life changes - new job, etc.  Also, I'm not 100% sure why I got listed as the repo owner here - like you said, I haven't contributed since June, but anyways let's see what we can do.

In terms of this specific issue, I'm not sure and I don't have a lab environment to run this one up in any way (see remark above about new job, I'm without a lot of toys and access at the moment) so I'm not sure what I can do to progress this one for you. The reality of a lot of these DSC resources is that they are supported by the community only, and as such getting specific fixes identified can take some time, especially when there isn't a real contributor to act as an owner.

I'll spend some time today catching up the PR's in this repo to help move those along, and if you can code up a fix for this issue I'm happy to commit to code reviewing it and getting it merged in. I'll do what I can to keep this repo warm until we can find someone to act as a more permanent owner on this one.

**Comment by Chirishman** [comment]
Thanks for the response! Yeah, looks like maybe the maintainer hat should pass to someone else. This is a recurrent issue where people's efforts to fix it keep fizzling. The problem is relevant to the following issues:

- #134 (this one)
- #27
- #26
- #59

The fix which I've included in my pull request ( #135 ) is more limited in scope than previous, not trying to add partition selection, merely filtering out recovery type partitions to fix the most basic potential scenario for this issue.

System and Reserved partitions don't break this code in the same way because though they are returned by Get-Partition they are not returned by Get-Volume.

**Comment by BrianFarnhill** [comment]
@Chirishman  OK so lets get #135 through (once you get the updates from PowerShell/dev the tests should pass and I'll review this week for you) and we can start to look at next steps for the other issues from there.
